modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/GcalEpics.scala

```diff
@@ -0,0 +1,138 @@
+
+    val nameAr = cs.map(_.getString("nameAr"))
+    def setArLampName(v: String): SeqAction[Unit] = setParameter(nameAr, v)
+
+    val stateAr = cs.map(_.getString("stateAr"))
+    def setArLampOn(v: Boolean): SeqAction[Unit] = setParameter(stateAr, v, toLampState)
+
+    val nameCuAr = cs.map(_.getString("nameCuAr"))
+    def setCuArLampName(v: String): SeqAction[Unit] = setParameter(nameCuAr, v)
+
+    val stateCuAr = cs.map(_.getString("stateCuAr"))
+    def setCuArLampOn(v: Boolean): SeqAction[Unit] = setParameter(stateCuAr, v, toLampState)
+
+    val nameIR = cs.map(_.getString("nameIR"))
+    def setIRLampName(v: String): SeqAction[Unit] = setParameter(nameIR, v)
+
+    val stateIR = cs.map(_.getString("stateIR"))
+    def setIRLampOn(v: Boolean): SeqAction[Unit] = setParameter(stateIR, v, toLampState)
+
+    val nameQH = cs.map(_.getString("nameQH"))
+    def setQHLampName(v: String): SeqAction[Unit] = setParameter(nameQH, v)
+
+    val stateQH = cs.map(_.getString("stateQH"))
+    def setQHLampOn(v: Boolean): SeqAction[Unit] = setParameter(stateQH, v, toLampState)
+
+    val nameXe = cs.map(_.getString("nameXe"))
+    def setXeLampName(v: String): SeqAction[Unit] = setParameter(nameXe, v)
+
+    val stateXe = cs.map(_.getString("stateXe"))
+    def setXeLampOn(v: Boolean): SeqAction[Unit] = setParameter(stateXe, v, toLampState)
+
+    val nameThAr = cs.map(_.getString("nameThAr"))
+    def setThArLampName(v: String): SeqAction[Unit] = setParameter(nameThAr, v)
+
+    val stateThAr = cs.map(_.getString("stateThAr"))
+    def setThArLampOn(v: Boolean): SeqAction[Unit] = setParameter(stateThAr, v, toLampState)
+  }
+
+  val state = epicsService.getStatusAcceptor("gcal::status")
+
+  def createLampAttribute(name: String, longName: String): EnumAttribute[BinaryOnOff] =
+    new EnumAttribute[BinaryOnOff](state, name + "LampState", name + "_LampState", longName + " lamp state")(classOf[BinaryOnOff])
+
+  val lampAr: EnumAttribute[BinaryOnOff] = createLampAttribute("Ar", "Argon")
+
+  val lampQH: EnumAttribute[BinaryOnOff] = createLampAttribute("QH", "Quartz Halogen")
+
+  val lampCuAr: EnumAttribute[BinaryOnOff] = createLampAttribute("CuAr", "Copper Argon")
+
+  val lampXe: EnumAttribute[BinaryOnOff] = createLampAttribute("Xe", "Xenon")
+
+  val lampThAr: EnumAttribute[BinaryOnOff] = createLampAttribute("ThAr", "Thorium Argon")
+
+  val lampIr: EnumAttribute[BinaryOnOff] = createLampAttribute("IR", "Infrared")
+
+  def shutter: Option[String] = Option(state.getStringAttribute("shutter").value)
+
+  def filter: Option[String] = Option(state.getStringAttribute("filter").value)
+
+  def diffuser: Option[String] = Option(state.getStringAttribute("diffuser").value)
+}
+
+object GcalEpics extends EpicsSystem {
+
+  val Log = Logger.getLogger(getClass.getName)
+  val CA_CONFIG_FILE = "/Gcal.xml"
+  val GCAL_TOP = "gc:"
+
+  private var instanceInternal = Option.empty[GcalEpics]
```

Comment:
Same comment as on the previous PR, I wish we had a cleaner way to initialize this